app/controllers/ReportListController.scala

```diff
@@ -56,17 +58,25 @@ class ReportListController(
   def extractReports = SecuredAction(WithPermission(UserPermission.listReports)).async { implicit request =>
```

Comment:
There is a dedicated error object MalformedQueryParams (AppError used for this purpose) that will return 400 HTTP code.

I would have done something like this :

    for {
      reportFilter <- ReportFilter
        .fromQueryString(request.queryString, request.identity.userRole)
        .liftTo[Future] {
          logger.warn(s"Failed to parse ReportFilter query params")
          throw MalformedQueryParams
        }
      _ = logger.debug(s"Parsing zone query param")
      zone <- (new QueryStringMapper(request.queryString))
        .timeZone("zone")
        .liftTo[Future] {
          logger.warn(s"Failed to parse zone query param")
          MalformedQueryParams
        }
      _ = logger.debug(s"Requesting report for user ${request.identity.email}")
      file <- asyncFileRepository
        .create(AsyncFile.build(request.identity, kind = AsyncFileKind.Reports))
      _ = reportsExtractActor ! ReportsExtractActor.ExtractRequest(file.id, request.identity, reportFilter, zone)
    } yield Ok